cmd/rukpakctl/cmd/content.go

```diff
@@ -0,0 +1,188 @@
+	}
+	url := bundle.Status.ContentURL
+
+	// Create a Job that reada from the URL and outputs contents in the pod log
+	mounttoken := true
+	job := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "rukpakctl-job",
+			Namespace: "default",
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:    "rukpakctl",
+							Image:   "curlimages/curl",
+							Command: []string{"sh", "-c", "curl -sSLk -H \"Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)\" -o - " + url + " | tar ztv"},
+						},
+					},
+					ServiceAccountName:           "rukpakctl-sa",
+					RestartPolicy:                "Never",
+					AutomountServiceAccountToken: &mounttoken,
+				},
+			},
+		},
+	}
+	_, err = opt.k8sclient.BatchV1().Jobs("default").Create(context.Background(), &job, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create a job: %+v", err)
+	}
+	option := metav1.DeletePropagationBackground
+
+	// Wait for Job completion
+	for {
+		deployedJob, err := opt.k8sclient.BatchV1().Jobs("default").Get(context.Background(), "rukpakctl-job", metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get a service account: %+v", err)
+		}
+		if deployedJob.Status.CompletionTime != nil {
+			break
+		}
+		time.Sleep(1 * time.Second)
+	}
+
+	// Get Pod for the Job
+	pods, err := opt.k8sclient.CoreV1().Pods("default").List(context.Background(), metav1.ListOptions{LabelSelector: "job-name=rukpakctl-job"})
+	if err != nil {
+		return fmt.Errorf("failed to find pods for job: %+v", err)
+	}
+	if len(pods.Items) != 1 {
+		return fmt.Errorf("there are more than 1 pod found for the job")
+	}
+
+	// Get logs of the Pod
+	logReader, err := opt.k8sclient.CoreV1().Pods("default").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).Stream(context.Background())
+	if err != nil {
+		return fmt.Errorf("failed to get pod logs: %+v", err)
+	}
+	defer logReader.Close()
+	if _, err := io.Copy(os.Stdout, logReader); err != nil {
+		return fmt.Errorf("failed to read log: %+v", err)
+	}
+
+	if err = opt.k8sclient.CoreV1().ServiceAccounts("default").Delete(context.Background(), "rukpakctl-sa", metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("failed to delete service account: %+v", err)
+	}
+	if err = opt.k8sclient.RbacV1().ClusterRoleBindings().Delete(context.Background(), "rukpakctl-crb", metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("failed to delete clusterrolebinding: %+v", err)
+	}
+	if err = opt.k8sclient.BatchV1().Jobs("default").Delete(context.Background(), "rukpakctl-job", metav1.DeleteOptions{PropagationPolicy: &option}); err != nil {
+		return fmt.Errorf("failed to delete job: %+v", err)
+	}
+	if err = opt.k8sclient.CoreV1().Pods("default").Delete(context.Background(), pods.Items[0].Name, metav1.DeleteOptions{}); err != nil {
+		return fmt.Errorf("failed to delete pod: %+v", err)
```

Comment:
alternatively, we could create the cluster role binding first, and then ownerref everything by it, so all these objects go away automatically. 

I believe creating a cluster role binding pointing to a non-existent SA is fine, on a further reconcile it will find the SA and succeed

Reply:
It's a good way for clean up.  I'll try it.

Reply:
It works well.  Tnanks!